virttest/libvirt_xml/vm_xml.py

```diff
@@ -705,11 +705,27 @@ def set_vm_vcpus(vm_name, value, current=None, virsh_instance=base.virsh):
                         % (current, value))
                 else:
                     vmxml['current_vcpu'] = current
+            topology = vmxml.get_cpu_topology()
+            if topology:
+                # Do change here with respect to vcpu
+                sockets = topology['sockets']
+                cores = topology['cores']
+                threads = topology['threads']
+                if (int(value) % int(threads)) != 0:
+                    # Do not raise test fail here,
+                    # lets leave it to libvirt to detect while define and start
+                    logging.error("vcpu: %s not the multiples of"
+                                  " threads: %s", value, threads)
+                new_cores = int(value)/(int(threads) * int(sockets))
```

Comment:
@will-Do Agreed, though the vcpus hotplug going to be core granularity, negative cases will have a probelmand API doing a calculation can also cause a issue later, changed now to take inputs from user.